### ISSUE TITLE
[IntelliJ][ReportedByUser] Uncaught Exception  java.lang.NullPointerException: Cannot invoke "Object.hashCode()" because "key" is null

### ISSUE THREAD

**Issue by krystian-panek-vmltech** [microsoft/azure-tools-for-java#7248]
IntelliJ build version: 2022.3.1 IU-223.8214.52
OS: Mac OS X
JDK: JetBrains s.r.o. 17.0.5
Plugin version: 3.72.0-2022.3
Additional Info: None
Parent component: 
```
com.intellij.openapi.wm.impl.IdeRootPane[,0,0,1512x944,layout=com.intellij.openapi.wm.impl.IdeRootPane$MyRootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=449,maximumSize=,minimumSize=,preferredSize=]
```
Error message: 
```
IdeaLoggingEvent[message=null, throwable=java.lang.NullPointerException: Cannot invoke "Object.hashCode()" because "key" is null
	at java.base/java.util.concurrent.ConcurrentHashMap.get(ConcurrentHashMap.java:936)
```

**Comment by wangmingliang-ms** [comment]
@krystian-panek-wttech @camelmann  @gerardv the provided stack trace is truncated causing us not able to find the root cause, can you share the full exception stack trace here if you encounter this error again?

**Comment by wangmingliang-ms** [comment]
@Flanker32 I guess the error should be related to the bicep language server client based on the limited information currently available.

**Comment by gerardv** [comment]
> @krystian-panek-wttech @camelmann @gerardv the provided stack trace is truncated causing us not able to find the root cause, can you share the full exception stack trace here if you encounter this error again?

I'm sorry but I can't reproduce the exception and cannot supply the full stack trace... This report was made by the IntelliJ Azure toolkit. I just submitted it so it can be examined.

**Comment by wangmingliang-ms** [comment]
@gerardv no worry, we will simply fix it based on the information currently available. Please add it here if you encounter this error again in the future.

**Comment by camelmann** [comment]
@wangmingliang-ms the error I get is related to the `exist` method in `BlobContainerClient`:
```
java.lang.NullPointerException: Cannot invoke "java.time.OffsetDateTime.withOffsetSameInstant(java.time.ZoneOffset)" because "dateTime" is null
	at com.azure.core.util.DateTimeRfc1123.toRfc1123String(DateTimeRfc1123.java:183)
	at com.azure.core.http.policy.AddDatePolicy$1.beforeSendingRequest(AddDatePolicy.java:33)
	at com.azure.core.http.policy.HttpPipelineSyncPolicy.lambda$process$0(HttpPipelineSyncPolicy.java:29)
	at reactor.core.publisher.MonoCallable.call(MonoCallable.java:92)
	at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:127)
	at reactor.core.publisher.MonoFlatMap.subscribeOrReturn(MonoFlatMap.java:53)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4475)
	at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:203)
	at reactor.core.publisher.MonoFlatMap.subscribeOrReturn(MonoFlatMap.java:53)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4475)
	at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:200)
	at reactor.core.publisher.MonoFlatMap.subscribeOrReturn(MonoFlatMap.java:53)
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:57)
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4490)
	at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:203)
	at reactor.core.publisher.MonoFlatMap.subscribeOrReturn(MonoFlatMap.java:53)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4475)
	at reactor.core.publisher.Mono.block(Mono.java:1741)
	at com.azure.storage.common.implementation.StorageImplUtils.blockWithOptionalTimeout(StorageImplUtils.java:129)
	at com.azure.storage.blob.BlobContainerClient.existsWithResponse(BlobContainerClient.java:263)
	at com.azure.storage.blob.BlobContainerClient.exists(BlobContainerClient.java:241)
```

**Comment by wangmingliang-ms** [comment]
> @wangmingliang-ms the error I get is related to the `exist` method in `BlobContainerClient`:
> 
> ```
> java.lang.NullPointerException: Cannot invoke "java.time.OffsetDateTime.withOffsetSameInstant(java.time.ZoneOffset)" because "dateTime" is null
> 	at com.azure.core.util.DateTimeRfc1123.toRfc1123String(DateTimeRfc1123.java:183)
> 	at com.azure.core.http.policy.AddDatePolicy$1.beforeSendingRequest(AddDatePolicy.java:33)
> 	at com.azure.core.http.policy.HttpPipelineSyncPolicy.lambda$process$0(HttpPipelineSyncPolicy.java:29)
> 	at reactor.core.publisher.MonoCallable.call(MonoCallable.java:92)
> 	at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:127)
> 	at reactor.core.publisher.MonoFlatMap.subscribeOrReturn(MonoFlatMap.java:53)
> 	at reactor.core.publisher.Mono.subscribe(Mono.java:4475)
> 	at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:203)
> 	at reactor.core.publisher.MonoFlatMap.subscribeOrReturn(MonoFlatMap.java:53)
> 	at reactor.core.publisher.Mono.subscribe(Mono.java:4475)
> 	at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:200)
> 	at reactor.core.publisher.MonoFlatMap.subscribeOrReturn(MonoFlatMap.java:53)
> 	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:57)
> 	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52)
> 	at reactor.core.publisher.Mono.subscribe(Mono.java:4490)
> 	at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:203)
> 	at reactor.core.publisher.MonoFlatMap.subscribeOrReturn(MonoFlatMap.java:53)
> 	at reactor.core.publisher.Mono.subscribe(Mono.java:4475)
> 	at reactor.core.publisher.Mono.block(Mono.java:1741)
> 	at com.azure.storage.common.implementation.StorageImplUtils.blockWithOptionalTimeout(StorageImplUtils.java:129)
> 	at com.azure.storage.blob.BlobContainerClient.existsWithResponse(BlobContainerClient.java:263)
> 	at com.azure.storage.blob.BlobContainerClient.exists(BlobContainerClient.java:241)
> ```

thanks @camelmann , this should be another problem, we'll investigate it.

**Comment by wangmingliang-ms** [comment]
this bug should have been fixed in the latest `3.73.0`, can please upgrade and try again?

**Comment by gerardv** [comment]
Will try, thank you.




________________________________
